pkg/monitor/datapath_debug.go

```diff
@@ -239,12 +240,31 @@ type DebugMsg struct {
 	Arg3    uint32
 }
 
-// DecodeDebugMsg will decode 'data' into the provided DebugMsg structure
-func DecodeDebugMsg(data []byte, dbg *DebugMsg) error {
-	return dbg.decodeDebugMsg(data)
+// Dump prints the message according to the verbosity level specified
+func (n *DebugMsg) Dump(args *api.DumpArgs) {
+	switch args.Verbosity {
+	case api.INFO:
+		// We don't print messages at INFO level
+		return
+	case api.JSON:
+		fmt.Fprintln(args.Buf, n.getJSON(args.CpuPrefix, args.LinkMonitor))
+	default:
+		fmt.Fprintf(args.Buf, "%s MARK %#x FROM %d DEBUG: %s\n", args.CpuPrefix, n.Hash, n.Source, n.Message(args.LinkMonitor))
+	}
+}
+
+// GetSrc retrieves the source endpoint for the message.
```

Comment:
In general, I'd advise to modify directly original commits with needed/suggested changes.